Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2050,12 +2051,20 @@ void CGContextDrawTiledImage(CGContextRef context, CGRect rect, CGImageRef image
         gradientStops[i].position = locations[i];
     }
 
+    if (options == kCGGradientDrawsAfterEndLocation) {
```

Comment:
Please add support for `kCGGradientDrawsBeforeStartLocation`.

I don't know what this is doing, actually:
It's setting the start to be at `1E-45`, and then setting the end (gradientCount) to be at .. 0?
So a gradient `0.1 0.2 0.3` would become: `1E-45 0.2 0.3 0` ??